jws/jws.go

```diff
@@ -197,20 +197,18 @@ func (jws Decoded) Verify() error {
 		return errors.New("malformed JWS header. alg and kid are required")
 	}
 
-	verificationMethodID := jws.Header.KID
-	verificationMethodIDParts := strings.Split(verificationMethodID, "#")
-	if len(verificationMethodIDParts) != 2 {
+	did, err := _did.Parse(jws.Header.KID)
+	if err != nil {
 		return errors.New("malformed JWS header. kid must be a DID URL")
 	}
 
-	var didURI = verificationMethodIDParts[0]
-
-	resolutionResult, err := dids.Resolve(didURI)
+	resolutionResult, err := dids.Resolve(did.URI)
 	if err != nil {
 		return fmt.Errorf("failed to resolve DID: %w", err)
 	}
 
-	verificationMethod, err := resolutionResult.Document.SelectVerificationMethod(didcore.ID(verificationMethodID))
+	vmSelector := didcore.ID(did.URL)
+	verificationMethod, err := resolutionResult.Document.SelectVerificationMethod(vmSelector)
```

Comment:
all of this logic to go from KID to the public key is a bit funky to me... perhaps we should expose something within the `did` (or `didcore`?) package which does all of this behind the scenes: input a KID and output a public key

idk just a thought, lmk if that resonates I'll add something to the backlog

Reply:
do you mean to go from a verification method id to the public key? or JWK KID?

Reply:
hmmmmmm https://github.com/TBD54566975/web5-go/issues/74